File: src/main/java/org/itmo/isLab1/dragoncaves/DragonCaveController.java
package org.itmo.isLab1.dragoncaves;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;
import org.itmo.isLab1.common.framework.CrudController;
import org.itmo.isLab1.dragoncaves.dto.*;

@RestController
@RequestMapping(value = "/api/dragon-caves", produces = MediaType.APPLICATION_JSON_VALUE)
public class DragonCaveController
    extends CrudController<
        DragonCave,
        DragonCaveDto,
        DragonCaveCreateDto,
        DragonCaveUpdateDto,
        DragonCaveService> {

    public DragonCaveController(
        DragonCaveService service
    ) {
        super(service);
    }
}

File: frontend/src/controllers/DragonCaveController.ts

import { DragonCaveDto } from '@/interfaces/dto/dragoncaves/DragonCaveDto';
import { DragonCaveCreateDto } from '@/interfaces/dto/dragoncaves/DragonCaveCreateDto';
import { DragonCaveUpdateDto } from '@/interfaces/dto/dragoncaves/DragonCaveUpdateDto';
import CrudController from '@/interfaces/crud/CrudController';


export default new CrudController<DragonCaveDto, DragonCaveCreateDto, DragonCaveUpdateDto>("dragon-caves")